astroid/inference_tip.py

```diff
@@ -41,6 +41,8 @@ def _inference_tip_cached(
         if result is None:
             raise UseInferenceDefault()
     except KeyError:
+        if len(_cache) > 127:
```

Comment:
If we get a single cache miss we clear the entire cache ? This does not feel right, I would expect that we clear the oldest entry and add the new one, in a kind of first in first out queue ?

Reply:
Easier than I thought to just make this an LRU-style cache.